lib/api/core/validation/index.js

```diff
@@ -1,690 +1,708 @@
+      if (field.multivalued.value) {
+        if (!Array.isArray(documentSubset[fieldName])) {
+          manageErrorMessage(field.path, errorMessages, 'The field must be multivalued, unary value provided.', verbose);
+          return false;
+        }
 
-  if (field.mandatory && !field.hasOwnProperty('defaultValue') && (!documentSubset.hasOwnProperty(fieldName) || typeof documentSubset[fieldName] === 'undefined' || documentSubset[fieldName] === null)) {
-    manageErrorMessage(field.path, errorMessages, 'The field is mandatory.', verbose);
-    return false;
-  }
+        if (field.multivalued.hasOwnProperty('minCount') && documentSubset[fieldName].length < field.multivalued.minCount) {
+          manageErrorMessage(field.path, errorMessages, `Not enough elements. Minimum count is set to ${field.multivalued.minCount}.`, verbose);
+          return false;
+        }
 
-  if (documentSubset.hasOwnProperty(fieldName) && typeof documentSubset[fieldName] !== 'undefined' && documentSubset[fieldName] !== null) {
-    if (field.multivalued.value) {
-      if (!Array.isArray(documentSubset[fieldName])) {
-        manageErrorMessage(field.path, errorMessages, 'The field must be multivalued, unary value provided.', verbose);
-        return false;
-      }
+        if (field.multivalued.hasOwnProperty('maxCount') && documentSubset[fieldName].length > field.multivalued.maxCount) {
+          manageErrorMessage(field.path, errorMessages, `Too many elements. Maximum count is set to ${field.multivalued.maxCount}.`, verbose);
+          return false;
+        }
 
-      if (field.multivalued.hasOwnProperty('minCount') && documentSubset[fieldName].length < field.multivalued.minCount) {
-        manageErrorMessage(field.path, errorMessages, `Not enough elements. Minimum count is set to ${field.multivalued.minCount}.`, verbose);
-        return false;
+        fieldValues = documentSubset[fieldName];
       }
+      else {
+        if (Array.isArray(documentSubset[fieldName])) {
+          manageErrorMessage(field.path, errorMessages, 'The field is not a multivalued field; Multiple values provided.', verbose);
+          return false;
+        }
 
-      if (field.multivalued.hasOwnProperty('maxCount') && documentSubset[fieldName].length > field.multivalued.maxCount) {
-        manageErrorMessage(field.path, errorMessages, `Too many elements. Maximum count is set to ${field.multivalued.maxCount}.`, verbose);
-        return false;
+        fieldValues = [documentSubset[fieldName]];
       }
 
-      fieldValues = documentSubset[fieldName];
-    }
-    else {
-      if (Array.isArray(documentSubset[fieldName])) {
-        manageErrorMessage(field.path, errorMessages, 'The field is not a multivalued field; Multiple values provided.', verbose);
-        return false;
+      if (this.types[field.type].allowChildren) {
+        nestedStrictness = this.types[field.type].getStrictness(field.typeOptions, strictness);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Split this 95 characters long line (which is greater than 80 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ALineLength)